src/bottom-bar/use-on-complete-callback.js

```diff
@@ -135,11 +139,26 @@ export default function useOnCompleteCallback() {
         useOnCompleteWhenValidNotRequiredClick()
     const onCompleteWithoutValidationClick =
         useOnCompleteWithoutValidationClick()
+    const hasCompulsoryDataElementOperandsToFillOut =
+        useHasCompulsoryDataElementOperandsToFillOut()
+    const setCompleteAttempted = useEntryFormStore(
+        (state) => state.setCompleteAttempted
+    )
 
     return () => {
         let promise
 
-        if (isLoading && offline) {
+        if (hasCompulsoryDataElementOperandsToFillOut) {
+            setCompleteAttempted(true)
```

Comment:
this is only set if there is a compulsory data element operand to fill out, so that invalid styling only gets applied if the compulsory data element operands need to be filled out (otherwise, they remain with just the `*`)